src/main/java/cloud/eppo/api/Attributes.java

```diff
@@ -61,4 +64,43 @@ public Attributes getCategoricalAttributes() {
   public Attributes getAllAttributes() {
     return this;
   }
+
+  /** Serializes the attributes to a JSON string, omitting attributes with a null value. */
+  public String serializeNonNullAttributesToJSONString() {
+    return serializeAttributesToJSONString(true);
+  }
+
+  @SuppressWarnings("SameParameterValue")
+  private String serializeAttributesToJSONString(boolean omitNulls) {
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode result = mapper.createObjectNode();
+
+    for (Map.Entry<String, EppoValue> entry : entrySet()) {
+      String attributeName = entry.getKey();
+      EppoValue attributeValue = entry.getValue();
+
+      if (attributeValue == null || attributeValue.isNull()) {
+        if (!omitNulls) {
+          result.putNull(attributeName);
+        }
+      } else {
+        if (attributeValue.isNumeric()) {
+          result.put(attributeName, attributeValue.doubleValue());
+          continue;
+        }
+        if (attributeValue.isBoolean()) {
+          result.put(attributeName, attributeValue.booleanValue());
+          continue;
+        }
+        // fall back put treating any other eppo values as a string
+        result.put(attributeName, attributeValue.toString());
+      }
+    }
```

Comment:
I don't think you need any of this. Jackson supports serializing maps to json, just pass that directly to `writeValueAsString`

Reply:
I'm not sure this would work, the value of the map is an `EppoValue`. Would Jackson just call `toString()` to get the property value? Does it then check to see whether it needs quotes or not? (e.g., if the EppoValue's `toString()` evaluated to `1` would that be a string or number to serialize? It seems to me it may be better to explicitly control this.

Reply:
im pretty sure it will just call `toString` for you

Reply:
it won't handle numbers vs. strings, though. everything will be treated as a string

Reply:
if you need that you'd have to write a custom deserializer

Reply:
Yeah and they'd have to then provide that deserializer do their call I think, so thinking we just leave the convenience method? It's also possible they don't even have Jackson at play in their codebase (only our package)

Reply:
we can have the deserializer as part of the library that we just add to the ObjectMapper on line 75, but this is simple enough that I'm ok keeping as is. if it ends up growing in complexity then I think having a serializer makes sense, but LGTM as is

Reply:
i dont know why I keep saying "deserializer" when I mean "serializer" 🤦🏽